smitebot: add working doctor command (checks + --json + docs) (#77)
morehouse/smite: smitebot/README.md

> # smitebot doctor

Reviewer: The first header should be `# smitebot` with an intro explaining what smitebot is.  Then probably `## Install`.  Then a `## Commands` subsection where `### smitebot doctor` lives.

`smitebot doctor` validates host prerequisites before running Smite campaigns.

## Install

Install `smitebot` once from this repository:

```bash
cargo install --path smitebot
```

After install, run it directly:

```bash
smitebot doctor
smitebot doctor --json
```

## Usage

```bash

> cargo run -p smitebot -- doctor
> cargo run -p smitebot -- doctor --json
> cargo run -p smitebot -- doctor --aflpp-path ~/AFLplusplus --smite-dir .
> ```

Reviewer: The usage here conflicts with the above `Install` section.  For simplicity:

```suggestion
```bash
smitebot doctor
smitebot doctor --json
smitebot doctor --aflpp-path ~/AFLplusplus --smite-dir .
```
```

## Checks

- `x86_64` architecture
- `/dev/kvm` is present and openable
- Docker daemon is reachable (`docker info`)
- AFL++ tools on `PATH` (`afl-fuzz`, `afl-cmin`, `afl-tmin`, `afl-whatsup`)
- Nyx packer `hget` exists under AFL++
- `libnyx.so` is found on `LD_LIBRARY_PATH`
- VMware backdoor is enabled
- Required Smite scripts are present and executable

## JSON output

> Right now, the doctor will print human readable output, with the `--json` flag, output will be in structured JSON:

Reviewer: ```suggestion
By default, output is in a human readable format. The `--json` flag changes output to structured JSON:
```

# smitebot

`smitebot` is the Smite automation CLI. It is intended to orchestrate common fuzzing workflows and reduce manual setup/operations.

## Install

Install `smitebot` once from this repository:

```bash
cargo install --path smitebot
```

After install, run it directly:

```bash
smitebot doctor
smitebot doctor --json
```

## Commands

### smitebot doctor

`smitebot doctor` validates host prerequisites before running Smite campaigns.

```bash
smitebot doctor
smitebot doctor --json
smitebot doctor --aflpp-path ~/AFLplusplus --smite-dir .
```

## Checks

- `x86_64` architecture
- CPU virtualization enabled (`vmx` or `svm`)
- `/dev/kvm` is present and openable

> - Docker daemon is reachable (`docker info`)

Reviewer 2: This needs to be updated now

- Required host tools (`bash`, `python3`)

> - AFL++ tools (`afl-fuzz`, `afl-cmin`, `afl-tmin`, `afl-whatsup`) available on `PATH` or under `--aflpp-path`

Reviewer 2: Missing host tool check, Also, let’s reorder this based on how we are actually checking in `doctor.rs`

- Nyx packer `hget` exists under AFL++
- `libnyx.so` is found on `LD_LIBRARY_PATH` or under `--aflpp-path`
- VMware backdoor is enabled
- Required Smite scripts are present and executable
- Required workload Dockerfiles are present

## JSON output

By default, output is in a human readable format. The `--json` flag changes output to structured JSON:

```json
{
  "checks": [
    { "name": "x86_64 architecture", "passed": true },

>     { "name": "Docker daemon reachable", "passed": false, "reason": "docker info exited with status exit status: 1" }

Reviewer 2: This needs to be updated now

# smitebot

`smitebot` is the Smite automation CLI. It is intended to orchestrate common fuzzing workflows and reduce manual setup/operations.

## Install

Install `smitebot` once from this repository:

```bash
cargo install --path smitebot
```

After install, run it directly:

```bash
smitebot doctor
smitebot doctor --json
```

## Commands

### smitebot doctor

`smitebot doctor` validates host prerequisites before running Smite campaigns.

```bash
smitebot doctor
smitebot doctor --json

> smitebot doctor --aflpp-path ~/AFLplusplus --smite-dir .

Reviewer 2: `smitebot doctor --aflpp-path ~/AFLplusplus --smite-dir .` is failing for me:
```text
[fail] AFL++ Nyx packer hget: ~/AFLplusplus/nyx_mode/packer/packer/linux_x86_64-userspace/bin64/hget not found or not executable
[fail] libnyx.so locatable: libnyx.so not found in LD_LIBRARY_PATH or --aflpp-path
```
However, passing the explicit path or using an environment variable works for me. I think `~` is not resolving correctly

Author: resolved in newest commit
